src/main/java/teammates/ui/controller/InstructorStudentRecordsPageAction.java

```diff
@@ -58,76 +54,59 @@ public ActionResult execute() throws EntityDoesNotExistException {
             }
         }
 
-        List<FeedbackSessionAttributes> feedbacks = logic.getFeedbackSessionsListForInstructor(account.googleId);
+        List<FeedbackSessionAttributes> sessions = logic.getFeedbackSessionsListForInstructor(account.googleId);
 
-        filterFeedbackSessions(courseId, feedbacks);
+        filterFeedbackSessions(courseId, sessions, instructor, student);
 
-        data.sessions = new ArrayList<SessionAttributes>();
-        data.sessions.addAll(feedbacks);
-        Collections.sort(data.sessions, SessionAttributes.DESCENDING_ORDER);
-        CommentAttributes.sortCommentsByCreationTimeDescending(data.comments);
+        Collections.sort(sessions, FeedbackSessionAttributes.DESCENDING_ORDER);
+        CommentAttributes.sortCommentsByCreationTimeDescending(comments);
+        
+        StudentProfileAttributes studentProfile = null;
 
-        if (data.student.googleId == "") {
+        if (student.googleId == "") {
             statusToUser.add(Const.StatusMessages.STUDENT_NOT_JOINED_YET_FOR_RECORDS);
-        } else if (!data.currentInstructor
-                        .isAllowedForPrivilege(data.student.section,
-                                               Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
+        } else if (!instructor.isAllowedForPrivilege(student.section,
+                                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
             statusToUser.add(Const.StatusMessages.STUDENT_PROFILE_UNACCESSIBLE_TO_INSTRUCTOR);
         } else {
-            data.studentProfile = logic.getStudentProfile(data.student.googleId);
-            Assumption.assertNotNull(data.studentProfile);
+            studentProfile = logic.getStudentProfile(student.googleId);
+            Assumption.assertNotNull(studentProfile);
         }
 
-        // call this function right before the 'no-records' check
-        loadStudentProfile();
```

Comment:
It does, but look at the few lines on top of this: they did _exactly the same thing_.
The same can be observed in StudentProfile before the migration some time back as well.